File: Data Structures & Algorithms/two-integer-sum/submission-1.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var map = new Dictionary<int,int>();

        for(var i = 0; i < nums.Length; i++) {
            var sub = target - nums[i];
            if(map.ContainsKey(sub))
                return new int[] {map[sub] , i};
            map.Add(nums[i] , i);
        }

        return new int[] {};
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-1.go
func combinationSum(nums []int, target int) [][]int {
	res := [][]int{}
	sub := []int{}

	var dfs func(int, int)

	dfs = func(i int, sum int) {
		if sum == target {
			temp := make([]int, len(sub))
			copy(temp, sub)
			res = append(res, temp)
			return
		}

		if i >= len(nums) || sum > target {
			return 
		}

		sub = append(sub, nums[i])
		dfs(i, sum+nums[i])
		sub = sub[:len(sub)-1]
		dfs(i+1, sum)
	}

	dfs(0, 0)

	return res
}
